openlibrary/macros/BookByline.html

```diff
@@ -5,15 +5,19 @@
   $ url = author.get('url')
   $ rendered_name = cond(name, truncate(name, 40), _("Unknown author"))
   $if show_years and author.get('birth_date'):
-    $ birth_date_formatted = extract_year(author.get('birth_date'))
+    $ birth_date_formatted = extract_year(author.get('birth_date'), int_only=False)
     $if author.get('death_date'):
-      $ death_date_formatted = extract_year(author.get('death_date'))
+      $ death_date_formatted = extract_year(author.get('death_date'), int_only=False)
     $else:
       $ death_date_formatted = ''
-  $if show_years and url and author.get('birth_date'):
+  $if show_years and url and author.get('birth_date') and author.get('death_date'):
```

Comment:
Then here we can reference the `birth_year` and `death_year` variables directly instead of relying on `author.get('birth_date')` implying that `$birth_date_formatted` exists, which is a bit confusing.

```suggestion
  $if show_years and url and birth_year and death_year:
```